ecosystem/sep-xxxx.md

```diff
@@ -0,0 +1,182 @@
+
+Our focus is Step 2: storing the asset within the account.
+
+### Encoding Binary Data
+We begin by describing how binary data should be represented in a `ManageData` entry.
+
+Since the _key_ is restricted specifically to printable characters (ref: Stellar Core's [`isString32Valid()`](https://github.com/stellar/stellar-core/blob/b031954458df3bb31d8d98f136c6fee40523a10d/src/util/types.cpp#L89-L101) and C's [`isprint()`](https://en.cppreference.com/w/cpp/string/byte/isprint)), this gives us an encoding space of 94 characters or ~6.5 bits per character (a ~25% reduction in the space). From some cursory research, the [BasE91] encoding scheme closest to this theoretical limit and outperforms base64 encoding.
+
+The _value_ can be any binary data; thus, no encoding is necessary.
+
+### Representing the Data
+With the encoding described [above](#encoding-binary-data), we can now discuss the actual storage. Again, we'll use the `ManageData` operation to store the binary data. 
+
+The first entry _may_ be a [MIME type], followed by the first chunk of data, separated by a dash character (`-`). The MIME type *must* fit in a single key, and so must be <= 63 characters long.
+
+If the MIME type is excluded, the dash _must_ still be included so that readers can easily distinguish where the type starts and the data begins.
+
+Every subsequent entry is simply the next BasE91-encoded chunk, sized to fit into the 64-length key string, and the next 64-byte binary chunk.
+
+For example,
+
+| Key | Value |
+|:----|:-----:|
+| text/plain-'ZP^gp@... | .. 64 bytes .. |
+| Io1Tc%ZEYkE#^IR`0e... | .. 64 bytes .. |
+| %yO){B&mA#_1:W8{lT... | .. 64 bytes .. |
+| Cvnu8jZBP>1]TX2$g%B'  | <empty> |
+
+would be a valid entry. There's no need for padding.
+
+#### Implementation Note
+Because you cannot predict the length of your encoded string without doing the encoding itself, it will be necessary to use trial-and-error to find the longest binary chunk that fits in <= 64 bytes. While this does mildly impact encoding time, it ensures optimality on-chain.
+
+(This is the function of `_encode_nearest()` in the sample implementation, [below](#implementation).)
+
+### Pricing Implications
+Every additional `ManageData` entry encurs an increase in the base reserve required by an account by 0.5 XLM. Some napkin-math pricing tables ensue:
+
+| Data Size | ManageData Entries | Cost (XLM) |
+|----------:|--------------------|------------|
+|  10 B     | 1                  | 0.5        |
+|   1 KB    | 9                  | 4.5        |
+|  50 KB    | 418                | 209        |
+| 112 KB    | 937                | 468.5      |
+|   1 MB    | 8358               | 4179       |
+
+Based on an approximation formula to account for encoding expanding the binary size by 10-15%:
+
+    ceil(|data| / ((64 / 1.15) + 64))
+
+Note that accounts are limited to 1000 sub-entries (so just over 100 KiB), so larger assets would need to either be split across multiple accounts or stored off-chain.
```

Comment:
It's unclear if multi-account NFTs is a feature of this SEP that isn't clearly specified, or if the SEP is stating that with extensions it could be supported.

Reply:
The intent is for it to be the latter; I will reword to clarify this.

However, do you think it (the former) belongs here? We could define a format for referencing sub-accounts, e.g. having a list of entries that are public keys that are actually data-holding accounts. Of course, that begs the question about the next size level, heh. We're approaching filesystem territory.

Reply:
My vote for this is definitely single account storage. Beyond that and it's likely you'll want to look external to Stellar for storage. If we want to specify multi account storage that should be a separate SEP imo as it would be quite the discussion on association. It's also not anything I've ever seen anyone try and attempt yet so the need isn't there.

Reply:
+1 The complexity climbs quickly, so I think there's value in keeping any multi-account setup out-of-scope of this SEP. So I think it'd be clearer if the SEP stated a max size. If anyone ever wants larger they could write another SEP I think?

Reply:
Yep, agreed here. The max size is difficult because it's contingent on the _other_ subentries in the account (e.g. signers); I do provide a rough approximation but I can make this a little more refined.